lib/matplotlib/image.py

```diff
@@ -552,7 +566,7 @@ def _make_image(self, A, in_bbox, out_bbox, clip_bbox, magnification=1.0,
                      cbook._setattr_cm(self.norm, vmin=s_vmin, vmax=s_vmax):
                     output = self.norm(resampled_masked)
             else:
-                if A.ndim == 2:  # _interpolation_stage == 'rgba'
+                if A.ndim == 2:  # interpolation_stage == 'rgba'
```

Comment:
Can we do this as an `elif` and add an `else` that raises?  Or do we do sufficient validation else where that we are sure it will always be one of the valid keys when we get here?

Reply:
Yeah, keys are validated elsewhere, and this needs to be an else because it covers both ndim>2 and rgba.